scriptworker 23.0.0 - remove gpg support

escapewindow opened the pull request.

escapewindow commented:
I managed to get this onto signing-linux-dev1 via my puppet env. It removed the cron entry, claimed a task, and went green.

tomprince commented on modules/scriptworker/manifests/chain_of_trust.pp:
```diff
@@ -32,47 +16,9 @@
-          path    => '/usr/local/bin/:/bin:/usr/sbin:/usr/bin',
-          user    => $username,
-          command => "find ${git_pubkey_dir} -type f | xargs md5sum | sort > ${basedir}/.git-pubkey-dir-checksum";
-  }
```
I think all the files referenced here (both directly, and via exec) probably want to be marked as absent?

escapewindow replied:
I thought about that. I leaned towards keeping them, and getting rid of the last traces by migrating to docker/GCP. Because they're no longer valid keys according to scriptworker, I'm less concerned here. I can change the patch to specify `absent`, but I'll have to keep the configs in place to do so.

tomprince replied:
It *looks* like all the files are fixed relative to the basedir, so you could hard-code the things to remove? I don't think anything actually uses the flexibility to specify them individually.

escapewindow replied:
Smart. https://grokbase.com/t/gg/puppet-users/1446kqrt5g/how-to-remove-a-directory#201404083akctk6dr7pmqf5cndxfste3ky worked!

tomprince commented:
There are also the the two entries in moco-config.pp to be removed.

escapewindow commented:
I was able to test file cleanup on signing-linux-dev1; worked! Added hiera cleanup to the bug checklist, and I addressed the moco-config.pp entries. It's too late to roll out today, so if this passes review, I'll probably aim for Thursday after we're done with releases. Thanks!
